state/application.go

```diff
@@ -474,6 +484,56 @@ func (op *DestroyApplicationOperation) destroyOps() ([]txn.Op, error) {
 	return ops, nil
 }
 
+func (op *DestroyApplicationOperation) unassignBranchOps() ([]txn.Op, error) {
+	m, err := op.app.st.Model()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	appName := op.app.doc.Name
+	branches, err := m.applicationBranches(appName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if len(branches) == 0 {
+		return nil, errors.NotFoundf("branch for %q", appName)
```

Comment:
Same as with the units, just return nil and remove the `NotFound` detection from the caller.

Reply:
done